### PR TITLE
[macOS] Hardcode PyPy 7.3.1 for macOS-10.13

### PR DESCRIPTION
**Pull request by miketimofeev**
# Description
The recently released PyPy 7.3.2 for High Sierra is broken and is already fixed in the main PyPy branch but not yet released https://foss.heptapod.net/pypy/pypy/-/issues/3311
To avoid build failures 7.3.1 can be installed for HS.
As an additional change — workaround for 7.3.1 removed from 10.14 & 10.15 PyPy installations.

#### Related issue:
https://github.com/actions/virtual-environments-internal/issues/1217

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/pypy.sh [hunk 1]
```diff
@@ -66,16 +66,20 @@ function InstallPyPy
...
 pypyVersions=$(curl -4 -s --compressed $uri | grep 'osx64' | awk -v uri="$uri" -F'>|<' '{print uri$5}')
-
 toolsetVersions=$(get_toolset_value '.toolcache[] | select(.name | contains("PyPy")) | .versions[]')
+versionPattern="v[0-9]+\.[0-9]+\.[0-9]+-"
+
+# PyPy 7.3.2 for High Sierra is broken, use 7.3.1 instead https://foss.heptapod.net/pypy/pypy/-/issues/3311
+if is_HighSierra; then
+    versionPattern="v7.3.1-"
+    # PyPy 7.3.1 relies on system libffi.6.dylib, which is not existed in in libffi 3.3 release. As a workaround symlink can be created
+    ln -s libffi.7.dylib /usr/local/opt/libffi/lib/libffi.6.dylib
```
Comment (maxim-lobanov) [inline images/macos/provision/core/pypy.sh:78]: Have you confirmed that this workaround is not required for PyPY 7.3.2 on Catalina / Mojave?

Comment (miketimofeev) [reply images/macos/provision/core/pypy.sh:78]: yep, PyPy 7.3.2 was installed without any issues

**Review by al-cheb — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Comment by EwoutH** [comment]
I think this hardcode can be removed, since PyPy v7.3.3 is released right?

**Comment by maxim-lobanov** [comment]
@EwoutH , this hardcode is only for High Sierra - pretty old image that is not available publicly anymore. Mojave and Catalina images should always install the latest PyPy

**Pull request merged**